### DISCUSSION TITLE
$user = Auth::user(); is getting null while changing route.

### DISCUSSION THREAD

**Discussion by ryaned2020** [Kyon147/laravel-shopify#129] [Q&A] [answered]
After installing the application, I can retrieve the authenticated user's information such as their shop domain and email, which is great. However, when I change routes and try to access auth:user, I get a null. It seems that the /authenticate endpoint isn't keeping the user in session.

Here is my route list. 
```
Route::get('/', function () {
    return view('welcome');
})->middleware(['verify.shopify'])->name('home');

Route::get('/about', [AboutController::class, 'index']);
```

Its the AboutController
```
class AboutController extends Controller
{
     public function index()
    {
      
        return json_encode(Auth::user());
    }
}
```

I want to get the user data there, but its giving me null data in Auth::user , like [screenshot](https://prnt.sc/I8BC8qWU69aE)

**Comment by Kyon147** [comment, marked as answer]
`verify.shopify` needs to be on every route, that is why you are losing the `Auth::user()` as your about route has no middleware attached to it.

**Comment by ryaned2020** [comment]
Yes, that's I tried too. when I am adding the middleware `verify.shop`  and try to go the route. its just showing the loading bar and nothing happening. Even if I want to click on `home ('/')` again its just showing loading. [screenshot](https://prnt.sc/ikDKA_VeBoAo)

For check the scenario when I am opening the `home('/')` via new tab, seems can not be redirected to the shop url. 
`https://undefined/admin/apps/xxxxxxxx/authenticate/token?shop=mainstorebd.myshopify.com&target=%2F`

1. web.php
```
Route::get('/', function () {
    return view('home');
})->middleware(['verify.shopify'])->name('home');



Route::get('/about', function () {
   return view('about');
})->middleware(['verify.shopify']);
````


2. home.blade.php
```
       @extends('shopify-app::layouts.default')
        <ul>
            <a href="/">Home</a>
            <a href="/about">about</a>
            <a href="/contact">contact</a>
        </ul>
        @section('content')
    <!-- You are: (shop domain name) -->
        <p>You are: {{ $shopDomain ?? Auth::user()->name }}</p>
        <p>You are: {{ Auth::user()->email }}</p>
        @endsection

       @section('scripts')
    @parent

    <script>
        actions.TitleBar.create(app, { title: 'Test App Template' });
    </script>
@endsection

```

**Comment by mdrathik** [comment]
@ryaned2020 
I think you should check the routing and linking documentation to better understand the process. Please go to this [link](https://github.com/Kyon147/laravel-shopify/wiki/Authentication-Process#links--routing) and read if you are using a spa or not. . Regarding the use of `<a href="/">Home</a>`,  I would suggest using `URL::tokenRoute('home')` instead.

As @Kyon147  has correctly pointed out, only the `verify.shopify` route will allow the retrieval of authorized user data.

**Comment by Kyon147** [comment]
@ryaned2020 as @mdrathik you need to make sure you are passing the `host` param on every request as well.

By using just normal `<a>` you are not passing the required params that Shopify need, so AppBridge is failing to load on your other route and you are seeng the loading screen that we try to init appBridge on.
